### PR TITLE
Refactor BOT Court page with MobX 

### PR DESCRIPTION
**Pull request by amazingandyyy**

### CODE REVIEW

**Comment by amazingandyyy** [comment]
This branch is based on branch of #722, please check and ship #722 first.

File: src/scenes/BotCourt/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  render() {
+    const { list, loadMore, loadingMore, loading } = this.props.store.botCourt;
+    if (loading) return <Loading />;
+    const events = (list || []).map((event, i) => <EventCard key={i} index={i} {...event} />); // eslint-disable-line
+    return (
+      <Fragment>
+        <TopActions />
+        {(events.length < 1) && <EventsEmptyBg />}
+        {(events.length > 1) && <InfiniteScroll
```
Comment (alex-cory) [inline src/scenes/BotCourt/index.js]: Please wrap this like we do in the [bodhi docs](https://bodhi.frontify.com/document/230177#/react-best-practices)
```js
{(events.length > 1) && (
  <InfiniteScroll
  />
)}
```

File: src/stores/activitiesStores/Finalize.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      this.loadingMore = true;
+      this.skip += this.limit;
+      const nextFewEvents = await this.fetch();
+      runInAction(() => {
+        this.list = [...this.list, ...nextFewEvents];
+        this.loadingMore = false;
+      });
+    }
+  }
+  fetch = async (limit = this.limit, skip = this.skip) => {
```
Comment (alex-cory) [inline src/stores/activitiesStores/Finalize.js]: space please :)

File: src/stores/activitiesStores/Finalize.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    this.hasMore = true;
+    this.app.ui.location = AppLocation.resultSetting;
+    this.list = await this.fetch(limit);
+    runInAction(() => {
+      this.skip += limit;
+      this.loading = false;
+    });
+  }
+  // load more
```
Comment (alex-cory) [inline src/stores/activitiesStores/Finalize.js]: space please :) (do we need this // load more comment)

File: src/stores/activitiesStores/Finalize.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  @observable loadingMore = false
+  @observable list = []
+  @observable hasMore = true
+  @observable skip = 0
+  limit = 50
+
+  constructor(app) {
+    this.app = app;
+  }
+  // init to fetch the list
```
Comment (alex-cory) [inline src/stores/activitiesStores/Finalize.js]: space please :)

**Comment by amazingandyyy** [comment]
too outdated will create a new PR.

**Pull request closed without merging**